reporting/src/main/java/org/wso2/testgrid/reporting/EmailReportProcessor.java

```diff
@@ -69,76 +69,76 @@ public EmailReportProcessor() {
-        Path filePath;
-        filePath = Paths.get(TestGridUtil.deriveTestIntegrationLogFilePath(testPlan));
-
-        if (Files.exists(filePath)) {
-            try (BufferedReader bufferedReader = Files.newBufferedReader(filePath)) {
-                String currentLine;
-                int lineCount = 0;
-                while ((currentLine = bufferedReader.readLine()) != null) {
-                    stringBuilder.append(currentLine).append(HTML_LINE_SEPARATOR);
-                    if (++lineCount == 100) {
-                        stringBuilder.append(".........").append(HTML_LINE_SEPARATOR).append(".........")
-                                .append(HTML_LINE_SEPARATOR).append("(view complete log in detailed results..)");
-                        break;
-                    }
-                }
-                return stringBuilder.toString();
-            } catch (IOException e) {
-                throw new ReportingException("Error occurred while reading integration-test log file to get" +
-                        " integration test results for test-plan with id " + testPlan.getId() + "of the product " +
-                        testPlan.getDeploymentPattern().getProduct().getName() + "having infra-combination as " +
-                        testPlan.getInfraParameters(), e);
-            }
-        } else {
-            if (testPlan.getStatus().equals(Status.SUCCESS) || testPlan.getStatus().equals(Status.FAIL)) {
-                logger.error("Integration-test log file does not exist at '" +
-                        Paths.get(TestGridUtil.getTestGridHomePath()).relativize(filePath) + "' for test-plan "
-                        + "with id '" + testPlan.getId() + "' of the product '" +
-                        testPlan.getDeploymentPattern().getProduct().getName() + "' having infra-combination as " +
-                        testPlan.getInfraParameters());
-                return "Integration test-log is missing. Please contact TestGrid administrator.";
-            } else {
-                return "Test status is " + testPlan.getStatus() + ". Hence can not display the log.";
-            }
+    private List<TestResult.TestCaseResult> getTrimmedTests(List<TestResult.TestCaseResult> tests,
+            int maxDisplayTestCount) {
+        final int actualCount = tests.size();
+        int displayCount = tests.size();
+        displayCount = displayCount < maxDisplayTestCount ? displayCount : maxDisplayTestCount;
+
+        tests = new ArrayList<>(tests.subList(0, displayCount));
+        if (displayCount < actualCount) {
+            TestResult.TestCaseResult continuation = new TestResult.TestCaseResult();
+            continuation.setClassName("...");
+            TestResult.TestCaseResult allTestsInfo = new TestResult.TestCaseResult();
+            allTestsInfo.setClassName("(view complete list of tests (" + actualCount + ") in testgrid-live..)");
+            tests.add(continuation);
+            tests.add(continuation);
```

Comment:
Duplicated line. Shall we remove that?

Reply:
This is done by purpose. I wanted the email to contain two '----' lines if the number of test cases are trimmed.